### ISSUE TITLE
Fix JSON parsing in json to struct function

### ISSUE THREAD

**Issue by cindyyuanjiang** [NVIDIA/spark-rapids#8204]
**Is your feature request related to a problem? Please describe.**
This is to follow up with json to struct function in https://github.com/NVIDIA/spark-rapids/pull/8174. The current JSON parsing in the implementation is having CUDF do name and type inference, which has issue with parsing and error handling. In particular, with parsing ints, if the int is too large or the format is not exactly correct, then CUDF will return a partial value (best effort to get something out) but spark is strict and will return a null.

**Describe the solution you'd like**
"We have put in a lot of work to make our cast code do the right thing when going from a string to an int, so if we ask CUDF to return Strings values instead of int values and then cast them afterwards we get the parsing that we want."

**Comment by andygrove** [comment]
@cindyyuanjiang @revans2 I think this issue can be closed now? When reading JSON we ask cuDF for strings for all primitive type and then cast to the primitive type in the plugin.

**Comment by revans2** [comment]
If the tests pass, then I am good with closing this.

**Comment by andygrove** [comment]
Ok, so there are still differences in how we handle types between from_json and json scan, so this is still valid. I will work on making these consistent.

**Comment by andygrove** [comment]
I think we will need this cuDF feature before we can resolve this issue:

https://github.com/rapidsai/cudf/issues/14830

**Comment by andygrove** [comment]
I propose that we make `GpuJsonToStruct` consistent with `GpuJsonScan` by reading top-level primitives as strings, then cast to the expected type, and fall back to CPU if the source JSON contains structs (until https://github.com/rapidsai/cudf/issues/14830 is implemented).

@revans2 Does that sound reasonable to you?

**Comment by revans2** [comment]
That sounds good to me.

**Comment by revans2** [comment]
The main part of the proposed fix is already covered by https://github.com/NVIDIA/spark-rapids/pull/10542 and follow on issues have been filed for remaining issues related to this.
